include/static-pages.php

```diff
@@ -122,23 +123,50 @@ public function page_link( $link, $id ) {
 	}
 
 	/**
-	 * Adds page_on_front and page_for_posts properties to the language objects.
+	 * Adds page_on_front and page_for_posts properties to language data before the object is created.
 	 *
-	 * @since 1.8
+	 * @since 3.4
+	 *
+	 * @param array $static_pages Array of language page_on_front and page_for_posts properties.
+	 * @param array $language Language data.
+	 * @return array {
+	 *     Static pages properties if exist.
 	 *
-	 * @param PLL_Language[] $languages The list of languages.
-	 * @param PLL_Model      $model     The instance of PLL_Model.
-	 * @return PLL_Language[]
+	 *     @type int $page_on_front  ID of the page on front in this language
+	 *     @type int $page_for_posts Id of the page for posts in this language
+	 * }
 	 */
-	public static function pll_languages_list( $languages, $model ) {
-		if ( 'page' === get_option( 'show_on_front' ) ) {
-			foreach ( $languages as $k => $language ) {
-				$languages[ $k ]->page_on_front = $model->post->get( get_option( 'page_on_front' ), $language );
-				$languages[ $k ]->page_for_posts = $model->post->get( get_option( 'page_for_posts' ), $language );
-			}
+	public function set_static_pages( $static_pages, $language ) {
+		$static_pages = $this->set_static_page( 'page_on_front', $static_pages, $language );
+		$static_pages = $this->set_static_page( 'page_for_posts', $static_pages, $language );
+
+		return $static_pages;
+	}
+
+	/**
+	 * Adds one of the page_on_front or page_for_posts properties to language data before the object is created.
+	 *
+	 * @since 3.4
+	 *
+	 * @param string $static_page  Static page option name; page_on_front or page_for_posts.
+	 * @param array  $static_pages Array of language page_on_front and page_for_posts properties.
+	 * @param array  $language Language data.
+	 * @return array {
+	 *     Static pages properties if exist.
+	 *
+	 *     @type int $page_on_front  ID of the page on front in this language
+	 *     @type int $page_for_posts Id of the page for posts in this language
+	 * }
+	 */
+	public function set_static_page( $static_page, $static_pages, $language ) {
```

Comment:
I suggest to make a method which gets a value instead of setting it. The calling method will modify the array. This should avoid you to pass eh second parameter and will make things more readable.

Reply:
Done in 742c81c7f973a383ddc60b601f74458de36042fc